### ISSUE TITLE
Apply PO token and Visitor data

### ISSUE THREAD

**Issue by clytras** [Mithronn/rusty_ytdl#43]
Could we make it so we can apply our PO token and Visitor Data?
There are a couple of ways to retrieve those values both in backend and in browsers, so it would be nice if we have the option to give and apply those values if we have them and just fallback back to iOS client by default if the experiment is detected.

~PS: I am getting 403 errors after dowloading some chunks using the latest version and it was working a couple days ago.
I think that if I can apply PO/VD tokens, such errors might be elliminated.~

**Comment by Mithronn** [comment]
I'd be happy to if you could do PR about this. I have no time to implement this feature right now

**Comment by clytras** [comment]
@Mithronn thanks for replying.

Regarding the PO/VD tokens, I will do my investigation and I will try to make it work, there are other libraries that already using those tokens. If I succeed I will send a PR.

My issue of the 403 errors is due to `request` mixed dependency. I have a `actix-web` proxy, that under the hood uses `request`. The proxy server works just fine, but when I add `rusty_ytdl` to my dependencies, the proxy server can't decode responses with an error `"error decoding response body"`. I tried to add more `reqwest` features on my library, like `brotli` but it does not work. If you know why this is happening I would appreciate a response that will guide me to the solution.

I have posted a SO question about it because I didn't want to spam you
https://stackoverflow.com/questions/78961802/rust-reqwest-proxy-server-dependency-issues-when-using-with-another-library
